storybookjs/storybook
# Interactions: Fix `waitFor` behavior while debugging

Issue: #16872

## What I did

Don't intercept calls inside callbacks, and bubble exceptions inside callbacks up to the parent rather than forwarding it to the next interceptable call. Show child interactions in log and display errors in the appropriate place. I got rid of the `forwardedException` concept entirely because it didn't (and can't) work with chained interactions (`a().b()`). Instead, any interaction that raises an exception will now appear in the log, even if it's not interceptable.

## How to test

1 - Install @storybook/testing-library0.0.14--canary.23.ac6b056.0 in the `oficcial-example` project
2 - Run `yarn start`

- [x] Is this testable with Jest or Chromatic screenshots? Yes
- [x] Does this need a new example in the kitchen sink apps? No

If your answer is yes to any of these, please make sure to include it in your PR.
